docs/src/cookbooks/cookbook.md

```diff
@@ -443,6 +444,42 @@ class UsingCloneTypeBundle[T <: Data](gen: T) extends Bundle {
 getVerilogString(new Top(new UsingCloneTypeBundle(UInt(8.W))))
 ```
 
+### <a name="bundle-unable-to-clone"></a> How do I deal with the "unable to clone" error?
+
+Most Chisel objects need to be cloned in order to differentiate between the software representation of the bundle field from its "bound" hardware representation, where "binding" is the process of generating a hardware component. For Bundle fields, this cloning is supposed to happen automatically with a compiler plugin.
+
+In some cases though, the plugin may not be able to clone the Bundle fields. The most common case for when this happens is when the `chisel3.Data` part of the Bundle field is nested inside some other data structure and the compiler plugin is unable to figure out how to clone the entire structure. It is best to avoid such nested structures.
+
+There are a few ways around this issue - you can try wrapping the problematic fields in Input(...), Output(...), or Flipped(...) if appropriate. You can also try manually cloning each field in the Bundle using the `chiselTypeClone` method in `chisel3.reflect.DataMirror`. Here's an example with the Bundle whose fields won't get cloned:
+
+```scala mdoc
+ import chisel3._
+
+ class CustomBundleBroken(elts: (String, Data)*) extends Record {
+ 	val elements = ListMap(elts.map {
+ 		case (field, elt) =>
+ 		field -> elt
+ 	}: _*)
+ 	def apply(elt: String): Data = elements(elt)
+ }
+```
+
+You can use `chiselTypeClone` to clone the elements as:
+
+
+```scala mdoc
+ import chisel3._
+ import chisel3.reflect.DataMirror
+ 
+ class CustomBundleBroken(elts: (String, Data)*) extends Record {
+ 	val elements = ListMap(elts.map {
+ 		case (field, elt) =>
+		field -> DataMirror.internal.chiselTypeClone(elt)
```

Comment:
is this what we want vs chiselTypeOf(...) ? I wish we could just settle on one API here. 

In other words, I don't want to actively encourage some weird pattern that people should never be doing in the first place.

Reply:
I believe this is the right one, I'm not sure if chiselTypeOf clones the type

Reply:
chiselTypeOf clones the type, it's the same as chiselTypeClone except it also checks that the type is bound as hardware. In this case, the elts will not be bound hardware so we need chiselTypeClone.

I agree we should settle on one API but they have different uses and figuring out how to not need both uses is subtle and beyond this PR.